api4/user.go

```diff
@@ -170,7 +170,13 @@ func getUserByEmail(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// No permission check required
+	// No permission check required, but still prevent users who can't see another user's email address from using this
+
+	sanitizeOptions := c.App.GetSanitizeOptions(c.IsSystemAdmin())
```

Comment:
Somewhat unrelated to this PR, but `sanitizeOptions` has a decidedly inverted name to the actual meaning. I'd have interpreted `sanitizeOptions["email"] == true` to mean that email should be sanitized, when in fact it maps to `o.PrivacySettings.ShowEmailAddress` and means that email can be shown.